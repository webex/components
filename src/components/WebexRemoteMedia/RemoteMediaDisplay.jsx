/* eslint-disable jsx-a11y/media-has-caption, react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';

import {useStream} from '../hooks';

/**
 * Remote Media Display component displays the meeting's remote video and audio html elements
 *
 * @param {object} props
 * @param {object} props.remoteAudio
 * @param {object} props.remoteShare
 * @param {object} props.remoteVideo
 * @returns {object} JSX of the component
 */
export default function RemoteMediaDisplay({remoteAudio, remoteShare, remoteVideo}) {
  const audioRef = useStream(remoteAudio);
  const videoRef = useStream(remoteVideo);
  const shareRef = useStream(remoteShare);

  return (
    <>
      {remoteVideo ? <video ref={videoRef} playsInline autoPlay /> : null}

      {remoteShare ? <video ref={shareRef} playsInline autoPlay /> : null}

      {remoteAudio ? <audio ref={audioRef} autoPlay /> : null}
    </>
  );
}

RemoteMediaDisplay.propTypes = {
  remoteAudio: PropTypes.object,
  remoteShare: PropTypes.object,
  remoteVideo: PropTypes.object,
};

RemoteMediaDisplay.defaultProps = {
  remoteAudio: null,
  remoteShare: null,
  remoteVideo: null,
};
