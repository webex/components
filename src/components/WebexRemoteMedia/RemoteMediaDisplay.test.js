import React from 'react';

import RemoteMediaDisplay from './RemoteMediaDisplay';

jest.mock('../hooks/useStream');

describe('Remote Media Display component', () => {
  describe('snapshot', () => {
    test('matches snapshot of empty component', () => {
      expect(shallow(<RemoteMediaDisplay />)).toMatchSnapshot();
    });

    test('matches snapshot of remote audio', () => {
      expect(shallow(<RemoteMediaDisplay remoteAudio={{}} />)).toMatchSnapshot();
    });

    test('matches snapshot of remote video', () => {
      expect(shallow(<RemoteMediaDisplay remoteVideo={{}} />)).toMatchSnapshot();
    });

    test('matches snapshot of remote share', () => {
      expect(shallow(<RemoteMediaDisplay remoteShare={{}} />)).toMatchSnapshot();
    });

    test('matches snapshot of remote audio and Video', () => {
      expect(shallow(<RemoteMediaDisplay remoteAudio={{}} remoteVideo={{}} />)).toMatchSnapshot();
    });

    test('matches snapshot of remote audio, video, and share', () => {
      expect(shallow(<RemoteMediaDisplay remoteAudio={{}} remoteVideo={{}} remoteShare={{}} />))
        .toMatchSnapshot();
    });
  });
});
