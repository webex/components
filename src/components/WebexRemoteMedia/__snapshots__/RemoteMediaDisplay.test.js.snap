// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remote Media Display component snapshot matches snapshot of empty component 1`] = `<Fragment />`;

exports[`Remote Media Display component snapshot matches snapshot of remote audio 1`] = `
<Fragment>
  <audio
    autoPlay={true}
  />
</Fragment>
`;

exports[`Remote Media Display component snapshot matches snapshot of remote audio and Video 1`] = `
<Fragment>
  <video
    autoPlay={true}
    playsInline={true}
  />
  <audio
    autoPlay={true}
  />
</Fragment>
`;

exports[`Remote Media Display component snapshot matches snapshot of remote audio, video, and share 1`] = `
<Fragment>
  <video
    autoPlay={true}
    playsInline={true}
  />
  <video
    autoPlay={true}
    playsInline={true}
  />
  <audio
    autoPlay={true}
  />
</Fragment>
`;

exports[`Remote Media Display component snapshot matches snapshot of remote share 1`] = `
<Fragment>
  <video
    autoPlay={true}
    playsInline={true}
  />
</Fragment>
`;

exports[`Remote Media Display component snapshot matches snapshot of remote video 1`] = `
<Fragment>
  <video
    autoPlay={true}
    playsInline={true}
  />
</Fragment>
`;
